src/EnergyPlus/WaterManager.cc

```diff
@@ -951,7 +956,51 @@ namespace WaterManager {
             }
             state.dataWaterData->RainFall.CurrentRate = schedRate * ScaleFactor / DataGlobalConstants::SecInHour; // convert to m/s
             state.dataWaterData->RainFall.CurrentAmount = state.dataWaterData->RainFall.CurrentRate * (TimeStepSys * DataGlobalConstants::SecInHour);
+        } else {
+            // placeholder: add EP checks for out of range precipitation value later -- yujie
+            // when there's no site:precipitation but non-zero epw precipitation, uset the epw precipitation as the CurrentRate
+            if (state.dataEnvrn->LiquidPrecipitation > 0.0) {
+                ShowWarningMessage(state, "Please be aware that precipitation depth in the .epw weather file is used in the RainCollector calculation as the site:precipitation object is missing. Please make sure the precipitation depth in the weather file is valid. Please refer to the 24-Hour Precipitation records by the State Climate Extremes Committee for a sanity check.");
+                // LiquidPrecipitation is for a certain timestep in an hour, the rate = depth / seconds in a timestep
+                state.dataWaterData->RainFall.CurrentRate = state.dataEnvrn->LiquidPrecipitation / (DataGlobalConstants::SecInHour / state.dataGlobal->NumOfTimeStepInHour);
+                // fixme: debug print
+//                fmt::print("{} {}-{} prec amount: dep={}, rate={}\n",
+//                           state.dataEnvrn->CurMnDy,
+//                           state.dataGlobal->HourOfDay,
+//                           state.dataGlobal->TimeStep,
+//                           state.dataEnvrn->LiquidPrecipitation,
+//                           state.dataWaterData->RainFall.CurrentRate);
+            } else {
+                // no site:precipitation, LiquidPrecipitation is zero but rain flag is on, assume 1.5mm rain
+                if (state.dataEnvrn->IsRain) {
+                    ShowWarningMessage(state, "Rain flag is on but precipitation in the weather file is missing, fill it with 1.5mm");
+                    state.dataWaterData->RainFall.CurrentRate = (1.5 / 1000.0) / DataGlobalConstants::SecInHour;
+                    state.dataWaterData->RainFall.CurrentAmount =
+                        state.dataWaterData->RainFall.CurrentRate * (TimeStepSys * DataGlobalConstants::SecInHour);
+                } else {
+                    state.dataWaterData->RainFall.CurrentRate = 0.0;
+                    state.dataWaterData->RainFall.CurrentAmount = 0.0;
+                }
+            }
+        }
+        state.dataWaterData->RainFall.CurrentAmount = state.dataWaterData->RainFall.CurrentRate * (TimeStepSys * DataGlobalConstants::SecInHour);
+        int month = std::stoi(state.dataEnvrn->CurMnDy.std::string::substr(0, 2));
```

Comment:
There has to be a better way than this to get the current month.  How about `state.dataEnvrn->Month`?

Reply:
Oh! Thanks for the pointer. I should have looked more for existing members.